identity/src/main/java/com/android/identity/CborUtil.java

```diff
@@ -0,0 +1,305 @@
+    private static boolean cborAreAllDataItemsNonCompound(@NonNull List<DataItem> items,
+                                                          @DiagnosticsFlags int flags) {
+        for (DataItem item : items) {
+            if ((flags & CborUtil.DIAGNOSTICS_FLAG_EMBEDDED_CBOR) != 0 && item.hasTag()) {
+                return false;
+            }
+            switch (item.getMajorType()) {
+                case ARRAY:
+                case MAP:
+                    return false;
+
+                default:
+                    // Do nothing
+                    break;
+            }
+        }
+        return true;
+    }
+
+    private static boolean cborFitsInSingleLine(@NonNull List<DataItem> items,
+                                                @DiagnosticsFlags int flags) {
+        return cborAreAllDataItemsNonCompound(items, flags) && items.size() < 8;
+    }
+
+    private static void toDiagnostics(@NonNull StringBuilder sb,
+                                      int indent,
+                                      @NonNull DataItem dataItem,
+                                      @DiagnosticsFlags int flags) {
+        int count;
+
+        boolean pretty = ((flags & DIAGNOSTICS_FLAG_PRETTY_PRINT) != 0);
+        String indentString = "";
+        if (pretty) {
+            StringBuilder indentBuilder = new StringBuilder();
+            for (int n = 0; n < indent; n++) {
+                indentBuilder.append(' ');
+            }
+            indentString = indentBuilder.toString();
+        }
+
+        if (dataItem.hasTag()) {
+            sb.append(String.format(Locale.US, "%d(", dataItem.getTag().getValue()));
+        }
+
+        switch (dataItem.getMajorType()) {
+            case INVALID:
+                sb.append("<invalid>");
+                break;
+
+            case UNSIGNED_INTEGER:
+                // Major type 0: an unsigned integer.
+                sb.append(((UnsignedInteger) dataItem).getValue());
+                break;
+
+            case NEGATIVE_INTEGER:
+                // Major type 1: a negative integer.
+                sb.append(((NegativeInteger) dataItem).getValue());
+                break;
+
+            case BYTE_STRING:
+                // Major type 2: a byte string.
+                byte[] bstrValue = ((ByteString) dataItem).getBytes();
+
+                if (dataItem.hasTag() && dataItem.getTag().getValue() == 24
+                        && (flags & DIAGNOSTICS_FLAG_EMBEDDED_CBOR) != 0) {
+                    sb.append("<< ");
+                    ByteArrayInputStream bais = new ByteArrayInputStream(bstrValue);
+                    List<DataItem> dataItems = null;
+                    try {
+                        dataItems = new CborDecoder(bais).decode();
+                        if (dataItems.size() >= 1) {
+                            if (dataItems.size() > 1) {
+                                Logger.w(TAG, "Multiple data items in embedded CBOR, "
```

Comment:
I think you should print all the items, or at least put something in the output saying that there are more items. The warning in the log might not be noticed.